internal/backend/connections/connections.go

```diff
@@ -16,14 +18,28 @@ import (
 type Connections struct {
 	fx.In
 
-	Z  *zap.Logger
-	DB db.DB
+	Z            *zap.Logger
+	DB           db.DB
+	Integrations sdkservices.Integrations
 }
 
 func New(c Connections) sdkservices.Connections { return &c }
 
 func (c *Connections) Create(ctx context.Context, conn sdktypes.Connection) (sdktypes.ConnectionID, error) {
-	conn = conn.WithNewID()
+	intg, err := c.Integrations.GetByID(ctx, conn.IntegrationID())
+	if err != nil {
+		return sdktypes.InvalidConnectionID, err
+	}
+
+	status := intg.Get().InitialConnectionStatus()
+	if !status.IsValid() {
```

Comment:
the integration status is not valid ?
or is it about a specific connection somehow?

Reply:
it means that initial connection status was not supplied.

Reply:
but how the integration knows which connection ?

Reply:
it doesn't care - this is initial status for a brand new unconfigured connection.